lib/src/ansi_code.dart

```diff
@@ -0,0 +1,278 @@
+// Copyright 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io' as io;
+
+/// The type of code represented by [AnsiCode].
+class AnsiCodeType {
+  /// The name of the type.
+  final String name;
+
+  /// A foreground color.
+  static const AnsiCodeType foreground = const AnsiCodeType._('foreground');
+
+  /// A style.
+  static const AnsiCodeType style = const AnsiCodeType._('style');
+
+  /// A background color.
+  static const AnsiCodeType background = const AnsiCodeType._('background');
+
+  /// A reset value.
+  static const AnsiCodeType reset = const AnsiCodeType._('reset');
+
+  const AnsiCodeType._(this.name);
+
+  @override
+  String toString() => 'AnsiType.$name';
+}
+
+/// Standard ANSI escape code.
```

Comment:
Maybe mention that this is just text-visuals, no console control or other fancy codes.